src/gleam/list.gleam

```diff
@@ -1814,3 +1815,40 @@ pub fn transpose(list_of_list: List(List(a))) -> List(List(a)) {
     }
   }
 }
+
+fn do_shuffle_pair_unwrap(list: List(#(idx_type, a)), acc: List(a)) -> List(a) {
+  case list {
+    [] -> acc
+    _ -> {
+      let [elem_pair, ..enumerable] = list
+      do_shuffle_pair_unwrap(enumerable, [elem_pair.1, ..acc])
+    }
+  }
+}
+
+fn do_shuffle_by_pair_indexes(
+  list_of_pairs: List(#(idx_type, a)),
+  compare: fn(idx_type, idx_type) -> Order,
```

Comment:
This value never changes so we don't need to pass it in as an argument

Reply:
Not yet, my idea was once we have optional args to let people supply better randomizers (crypographically better etc), and if they do they might need a different comparison function. But I will remove it. I can add it later.